src/init.gradle

```diff
@@ -475,3 +486,171 @@ abstract class LicenseReportText extends DefaultTask {
         println "Wrote $txtFile"
     }
 }
+
+
+abstract class DependencyTree extends DefaultTask{
+
+    private static final String[] excludedDeps = ["jackson-annotations", "bom"]
+    private static final String[] configOptions = ["allCodeCoverageReportClassDirectories", "compileClasspath"]
+    private Set<String> usableDependencies = []
+
+    @InputFile
+    abstract RegularFileProperty getLicenseReport()
+
+    @OutputFile
+    abstract RegularFileProperty getDependencyTreeReport()
+
+    void printDependencyTree(){
+        try {
+            Configuration config = findConfigForProject()
```

Comment:
:x: The configuration should be an _input_ of the task. Otherwise any change in the dependencies will result in the task being up-to-date. In addition, calling `project` at task execution time is going to become an error in future Gradle releases.

This means you need an input like:

```
@Input
abstract Property<ResolvedComponentResult> getRootComponent()
```

and set it when you configure the task. e.g look at how it's done in https://github.com/adammurdoch/dependency-graph-as-task-inputs/blob/main/plugins/src/main/java/ReportDependencyGraphTask.java#L20